test/new-e2e/tests/containers/base_test.go

```diff
@@ -575,3 +575,116 @@ func (suite *baseSuite[Env]) testEvent(args *testEventArgs) {
 		}, 2*time.Minute, 10*time.Second, "Failed finding `%s` with proper tags and message", prettyEventQuery)
 	})
 }
+
+type testHostTags struct {
+	ExpectedTags *[]string
+	OptionalTags *[]string
+}
+
+func sendEvent[Env any](suite *baseSuite[Env], alertType, text string, args *testHostTags) {
```

Comment:
This function seems to be specific to “host tags tests” based on the title that it generates line 589.
Maybe this should be reflected in the function name.
Or we should make it fully generic, add a `Title` parameter and use it for the functions asserting metrics and logs too.